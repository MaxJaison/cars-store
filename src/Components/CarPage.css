.wrapper {
  width: 800px;
  margin: 0 auto;
  padding: 24px;
}
