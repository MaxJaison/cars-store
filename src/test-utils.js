import React from "react";
import { render as rtlRender } from "@testing-library/react";
import { applyMiddleware, createStore } from "redux";
import { Provider } from "react-redux";
import reducer from "./reducers";
import thunk from "redux-thunk";
import { BrowserRouter } from "react-router-dom";

function render(
  ui,
  {
    initialState,
    store = createStore(reducer, applyMiddleware(thunk)),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    return (
      <Provider store={store}>
        <BrowserRouter>{children}</BrowserRouter>
      </Provider>
    );
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

export * from "@testing-library/react";
export { render };
