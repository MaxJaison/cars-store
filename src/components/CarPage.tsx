import React, { useEffect } from "react";
import { useDispatch, useSelector } from "react-redux";
import { Car, RootState } from "../types";
import { useParams } from "react-router-dom";
import { fetchCar } from "../actionCreators/CarsActions";
import "./CarPage.css";

interface ParamTypes {
  id: string;
}

const CarPage = () => {
  let { id } = useParams<ParamTypes>();
  const car: Car = useSelector((state: RootState) => state.car);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchCar(id));
  }, [dispatch, id]);

  return (
    <>
      {!Object.keys(car).length ? (
        <div>Loading car...</div>
      ) : (
        <div className="wrapper">
          <h2>
            {car.manufacturerName} {car.modelName}
          </h2>
          <p>
            Stock # {car.stockNumber} - {car.mileage.number / 1000}{" "}
            {car.mileage.unit} - {car.fuelType} - {car.color}
          </p>
          <p>
            This car is currently available and can be delivered as soon as
            <br />
            tomorrow morning. Please be aware that delivery times shown in
            <br />
            this page are not definitive and may change due to bad weather
            <br />
            conditions.
          </p>
        </div>
      )}
    </>
  );
};

export default CarPage;
