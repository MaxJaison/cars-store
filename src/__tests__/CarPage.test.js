import { render } from "../test-utils";
import React from "react";
import CarPage from "../components/CarPage";

test("renders CarPage", () => {
  const { container } = render(<CarPage />, {});
  expect(container).toMatchSnapshot();
});
