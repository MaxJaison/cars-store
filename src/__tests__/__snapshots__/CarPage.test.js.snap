// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CarPage 1`] = `
<div>
  <div>
    Loading car...
  </div>
</div>
`;
